vars/retrievePreviousBuild.groovy

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+/** Library to call build.sh script (builds OpenSearch and OpenSearch Dashboards distribution)
+ @param Map args = [:] args A map of the following parameters
+ @param args.inputManifest <required> - Relative path to input manifest containing all the components to build.
+ @param args.platform <required> - Platform of previous build to retrieve.
+ @param args.architecture <required> - Architecture of previous build to retrieve.
+ @param args.distribution <required> - Distribution of previous build to retrieve.
+ @param args.incremental <required> - String value contains the build number of previous build or the latest.
+ */
+void call(Map args = [:]) {
+    def lib = library(identifier: 'jenkins@jenkins_incremental', retriever: legacySCM(scm))
+    def inputManifestObj = lib.jenkins.InputManifest.new(readYaml(file: args.inputManifest))
+
+    def DISTRIBUTION_JOB_NAME = "${JOB_NAME}"
+    def revision = inputManifestObj.build.version
+
+    def DISTRIBUTION_PLATFORM = args.platform
+    def DISTRIBUTION_ARCHITECTURE = args.architecture
+    def distribution = args.distribution
+    def prefixPath = "${WORKSPACE}/download"
+    def DISTRIBUTION_BUILD_NUMBER
+
+    if (args.incremental.equalsIgnoreCase("latest")) {
```

Comment:
Can this arg `incremental` being changed to `version`? As the value is either the version or `latest`.

Reply:
or maybe be `incrementally_build_id`

Reply:
But this is the argument we would get from `buildManifest` groovy library. It could also be `false` there not just a simple id.

Reply:
Since empty/null/false are all same in this case, `previous_build_id` seems good fit. How about replacing `false` with `none/null` here? or @peterzhuamazon may have better ideas?

Reply:
I would like to keep it as the original `incremental` name for the argument. 
1. This argument will be initiated in the jenkinsfile and incremental is straightforward stating what this parameter is for.
2. This parameter will work as a dynamic option for user: enable/disable the incremental build process; if enable, give user option to choose what particular build artifacts to download. Since using Jenkins dynamic parameter is a little complicated, I lean on to name it more generic. 
What do you think? @gaiksaya @peterzhuamazon

Reply:
You can add that customization in main distribution file. When it comes to this specific library, you need a previous build id which needs to be mentioned explicitly. Still leaning toward `previous_build_id` for this specific library. How you showcase that at the distribution pipeline level can be different that this.

Reply:
Updated the PR with two separate parameters. One is a boolean for whether we enable incremental build; this will be used in `buildManifest` lib to determine whether trigger incremental build. The other is the generic one build ID to download previous build used in `retrievePreviousBuild` library. Please take a look. Thanks. @gaiksaya @peterzhuamazon